modules/react/banner/lib/BannerAction.tsx

```diff
@@ -0,0 +1,42 @@
+/** @jsx jsx */
+import {jsx} from '@emotion/core';
+import * as React from 'react';
+
+import {ExtractProps, createComponent, useModelContext} from '@workday/canvas-kit-react/common';
+import {Box} from '@workday/canvas-kit-labs-react/common';
+import {CSSProperties} from '@workday/canvas-kit-react/tokens';
+
+import {BannerModelContext} from './Banner';
+import {BannerModel, useBannerAction} from './hooks';
+
+export interface BannerActionProps extends ExtractProps<typeof Box, never> {
+  model?: BannerModel;
+  /**
+   * The text of the Banner action.
+   * @default 'View All'
+   */
+  children?: React.ReactNode;
+}
+
+const styles: CSSProperties = {
+  textDecoration: 'underline',
+};
+
+export const BannerAction = createComponent('span')({
+  displayName: 'Banner.Action',
+  Component: ({children = 'View All', model, ...elemProps}: BannerActionProps, ref, Element) => {
+    const localModel = useModelContext(BannerModelContext, model);
+    const props = useBannerAction(localModel, elemProps, ref);
+
+    return (
+      <Box
+        display={localModel.state.isSticky ? 'none' : 'inline'}
+        as={Element}
+        css={[styles]}
+        {...props}
+      >
+        {children}
+      </Box>
+    );
+  },
```

Comment:
Mostly curious here, i notice in the stories that the `onClick` is added to the top level component `Banner` is there a reason why this component isn't a `Tertiary` button or a button by default? Is the reasoning because the whole alert is clickable?

I figured if I were consuming this I would want to add the `onClick` the `Banner.Action` component, and the wrapping component just handles the model. Thoughts?

Reply:
> Is the reasoning because the whole alert is clickable?
Yes. I don't think it makes sense to have more than one click handler. Open to suggestion here.